test/typescript/exposed.test.ts

```diff
@@ -0,0 +1,14 @@
+/** Exercise exposed types/imports for different tsconfig esmoduleinterop settings */
+
+/* esModuleInterop: false (default), allowSyntheticDefaultImports: false (default) */
+import i18next, { Interpolator, Modules, WithT } from 'i18next';
+
+i18next.init();
```

Comment:
In a commonjs enviroment (node.js where `esModuleInterop` is not enabled), this will give a runtime error because the entry point doesn't provide a `module.exports.default` 

As you can check by running this codesandbox: https://codesandbox.io/s/node-typescript-h3gcd
``i18next.init({});
        ^
TypeError: Cannot read property 'init' of undefined
    at Object.<anonymous> (/sandbox/src/index.ts:4:9)```

Typescript does not use the index.js in the root of the directory. it uses the published `dist/cjs/i18next.js` which has this as export
```var i18next = new I18n();

module.exports = i18next;```

Reply:
tsc will use whatever file is pointed to in the main of the package.json, which I now see is different than I expected:

https://unpkg.com/i18next@17.0.6/package.json

I'll do a local build and see if I can correct it.

Reply:
Makes sense. Thank you for taking a look on this.

Reply:
the esm only has

```
var i18next = new I18n();
export default i18next;
```

whereas the cjs only has

```
var i18next = new I18n();
module.exports = i18next;
```

So named export is incorrect, and makes importing it differ depending the `esModuleInterop` setting again.  The _ideal_ solution to this is to *remove* the runtime **default** export and standardize on a named import only.  This would provide identical import uses no matter it being js, ts (any config permutation)

So, with that said, I have limited the ts export to named only, and added a named export to the index.js.

With that, the cjs generated is now:

```ts
var i18next = new I18n();

exports.i18next = i18next;
exports.default = i18next;
```

and esm:

```ts
var i18next = new I18n();

export default i18next;
export { i18next };
```

Reply:
TL;DR So with AND without the esModuleInterop, the import is always the same now: `import {i18next} from 'i18next'`.